### ISSUE TITLE
Training format from train_ner.py script not accepted by CLI - ValueError: Expected object or value

### ISSUE THREAD

**Issue by erotavlas** [explosion/spaCy#3723]
Very confused as to what training format to use for the CLI.
 
I've created a lot of training data in the format as required for the train_ner.py script here
https://github.com/explosion/spaCy/blob/master/examples/training/train_ner.py

Which worked fine and I was able to train models using that data.

```
[
    ("Who is Shaka Khan?", {"entities": [(7, 17, "PERSON")]}),
    ("I like London and Berlin.", {"entities": [(7, 13, "LOC"), (18, 24, "LOC")]}),
]
```

However when I tried to use this same data in the CLI (i.e with the above sample data provided in the train_ner.py script)

`python -m spacy train en out F2.json F2.json -p 'ner'`

In spacy 2.1.3 the output looks like this
```

(py36) bash-3.2$ python -m spacy train en out F2.json F2.json -p 'ner'
Training pipeline: ['ner']
Starting with blank model 'en'
Counting training words (limit=0)
{"entities": [(7, 17, "PERSON")]}
Traceback (most recent call last):
  File "/Users/user/anaconda3/envs/py36/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/Users/user/anaconda3/envs/py36/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/Users/user/anaconda3/envs/py36/lib/python3.6/site-packages/spacy/__main__.py", line 35, in <module>
    plac.call(commands[command], sys.argv[1:])
  File "/Users/user/anaconda3/envs/py36/lib/python3.6/site-packages/plac_core.py", line 328, in call
    cmd, result = parser.consume(arglist)
  File "/Users/user/anaconda3/envs/py36/lib/python3.6/site-packages/plac_core.py", line 207, in consume
    return cmd, self.func(*(args + varargs + extraopts), **kwargs)
  File "/Users/user/anaconda3/envs/py36/lib/python3.6/site-packages/spacy/cli/train.py", line 196, in train
    corpus = GoldCorpus(train_path, dev_path, limit=n_examples)
  File "gold.pyx", line 112, in spacy.gold.GoldCorpus.__init__
  File "gold.pyx", line 123, in spacy.gold.GoldCorpus.write_msgpack
  File "gold.pyx", line 163, in read_tuples
  File "gold.pyx", line 321, in read_json_file
  File "gold.pyx", line 378, in _json_iterate
  File "gold.pyx", line 375, in spacy.gold._json_iterate
  File "/Users/user/anaconda3/envs/py36/lib/python3.6/site-packages/srsly/_json_api.py", line 37, in json_loads
    return ujson.loads(data)
ValueError: Expected object or value
```

How do I use the data I created for the train_ner.py script in the CLI?
Or is there something else wrong here?

spaCy (2.1.3 and 2.0.18)
Python 3.6.8 :: Anaconda, Inc.
macOS Sierra 10.12.6

**EDIT:**

Later I found that the CLI only accepts the spacy json format as described here 
https://spacy.io/api/annotation#json-input

However according to another issue here, we are supposed to be able to convert the other train data format to the spacy json format using the convert command.  But I was unable to get it to work.  

An example of what the jsonl format looks like would be helpful.

**Comment by Arakkun** [comment]
Remember to prettify the json somewhere. 
Since spacy uses the dictionary keys to know the various things, you can add additional infos  in the json.
https://puu.sh/DqN2e/3c9d1c48b0.json
This is and excerpt
https://puu.sh/DqN7u/0068961ab0.png
as for NER, i guess the bare minimum would need tokenization (id and orth) plus ner.
So, in your case, the token part would be like
token:[
 {
  "id":0
  "orth":"Who"
  "ner":"O"
 }
 {
  "id":1
  "orth":"is"
  "ner":"O"
 }
 {
  "id":2
  "orth":"Shaka"
  "ner":"B-PERSON"
 }
 {
  "id":3
  "orth":"Khan"
  "ner":"L-PERSON"
 }
 {
  "id":4
  "orth":"?"
  "ner":"O"
 }
]

Basically, spacy train format uses both knowledge of tokens and BILUO+NER.
BILUO means you add
"B-"egin (if entity covers two tokens or more, the token where it Begin)
"I-"nner (if entity covers three tokens or more, the tokens before the Last and after the First)
"L-"ast (if entity covers two tokens or more, the token where it End)
"U-"nit (if entity covers only one token)
with "O" if the token isn't part of an entity

**Comment by ines** [comment]
> However according to another issue here, we are supposed to be able to convert the other train data format to the spacy json format using the convert command. But I was unable to get it to work.

Sorry if this was a bit confusing. The thing is, the "training data format" used in the Python training script examples is pretty arbitrary – it's not set in stone. If you look at the code, you'll see that it's calling `nlp.update` and that `nlp.update` takes a text and a dictionary with the annotations. So a logical way to store this is in a list of `(text, annotations)` tuples. However, you could also store this in any other way – like, a dictionary or a dataframe or whatever you prefer. As long as you call `nlp.update` with the corret values, it doesn't matter.

The `spacy convert` command is intended to convert other data formats to spaCy's format – for instance, `.conll`, which is how the Universal Dependencies treebanks etc. are distributed. This is mostly relevant for larger corpora that follow a different standard format.
